tests/system/test_asyncpg_connection.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import asyncpg
+import pytest
+import sqlalchemy
+from sqlalchemy.ext.asyncio import create_async_engine
+
+from google.cloud.alloydb.connector import AsyncConnector
+
+
+@pytest.mark.asyncio
+async def test_connection_with_asyncpg() -> None:
+    async def getconn() -> asyncpg.Connection:
+        async with AsyncConnector() as connector:
```

Comment:
we should swap the order of these two lines... the reason being for users copy/pasting this code.

`getconn` is called everytime the connection pool needs a fresh connection. Thus we are creating a new `AsyncConnector` for each connection instead of re-using the same one for each connection. Switching the order of these two lines will re-use the same `AsyncConnector` for all connections which is how we want users to be leveraging the Python Connector at scale to avoid quota issues.

Reply:
Makes perfect sense. This is a copy-paste of Cloud SQL's test. We'll need to make the same change there.